src/display/annotation_layer.js

```diff
@@ -440,31 +439,87 @@ class AnnotationElement {
    * Create quadrilaterals from the annotation's quadpoints.
    *
    * @private
-   * @param {boolean} ignoreBorder
    * @memberof AnnotationElement
-   * @returns {Array<HTMLElement>} An array of section elements.
    */
-  _createQuadrilaterals(ignoreBorder = false) {
-    if (!this.data.quadPoints) {
-      return null;
-    }
-
-    const quadrilaterals = [];
-    const savedRect = this.data.rect;
-    let firstQuadRect = null;
-    for (const quadPoint of this.data.quadPoints) {
-      this.data.rect = [
-        quadPoint[2].x,
-        quadPoint[2].y,
-        quadPoint[1].x,
-        quadPoint[1].y,
+  _createQuadrilaterals() {
+    if (!this.container) {
+      return;
+    }
+    const { quadPoints } = this.data;
+    if (!quadPoints) {
+      return;
+    }
+
+    const [rectBlX, rectBlY, rectTrX, rectTrY] = this.data.rect;
+    const width = rectTrX - rectBlX;
+    const height = rectTrY - rectBlY;
+
+    if (quadPoints.length === 1) {
+      const [, { x: trX, y: trY }, { x: blX, y: blY }] = quadPoints[0];
+      if (
+        width === trX - blX &&
+        height === trY - blY &&
+        rectBlX === blX &&
+        rectTrY === trY
+      ) {
```

Comment:
Can we directly compare the coordinates instead, since that feels more intuitive and readable?
```suggestion
      if (
        blX === rectBlX &&
        blY === rectBlY &&
        trX === rectTrX &&
        trY === rectTrY
      ) {
```